go/protocol/chat1/common.go

```diff
@@ -234,6 +234,18 @@ type MessageClientHeader struct {
 	OutboxInfo   *OutboxInfo              `codec:"outboxInfo,omitempty" json:"outboxInfo,omitempty"`
 }
 
+type MessageClientHeaderVerified struct {
+	Conv         ConversationIDTriple     `codec:"conv" json:"conv"`
+	TlfName      string                   `codec:"tlfName" json:"tlfName"`
+	TlfPublic    bool                     `codec:"tlfPublic" json:"tlfPublic"`
+	MessageType  MessageType              `codec:"messageType" json:"messageType"`
+	Prev         []MessagePreviousPointer `codec:"prev" json:"prev"`
+	Sender       gregor1.UID              `codec:"sender" json:"sender"`
+	SenderDevice gregor1.DeviceID         `codec:"senderDevice" json:"senderDevice"`
```

Comment:
`Sender`+`SenderDevice` are already checked inside `unboxV1` here https://github.com/keybase/client/blob/master/go/chat/boxer.go#L269. So either way.

`Prev` yes, good idea. You'll probably want to make sure that history actually upholds that invariant. I've been doing that by running new checks against my chats and reading with the gui/api.

The only fields left after that right now are `OutboxInfo` and `OutboxID`.